main.go

```diff
@@ -0,0 +1,242 @@
+	var b struct{ Data build }
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return build{}, err
+	}
+	if err := json.Unmarshal(body, &b); err != nil {
+		return build{}, errors.Wrap(err, string(body))
+	}
+	return b.Data, nil
+}
+
+func (cfg config) getBuilds(f filter) (builds, error) {
+	url := baseURL + "/apps/" + cfg.AppSlug + "/builds"
+
+	f.SortBy = "created_at"
+	queryParams, err := query.Values(f)
+	if err != nil {
+		return nil, err
+	}
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return nil, err
+	}
+	req.URL.RawQuery = queryParams.Encode()
+	req.Header.Add("accept", "application/json")
+	req.Header.Add("Authorization", string(cfg.AccessToken))
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	var builds struct {
+		Data []build `json:"data"`
+	}
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	if err := json.Unmarshal(body, &builds); err != nil {
+		return nil, errors.Wrap(err, string(body))
+	}
+	return builds.Data, nil
+}
+
+type filter struct {
+	Before           int    `url:"before,omitempty"`
+	Branch           string `url:"branch,omitempty"`
+	SortBy           string `url:"sort_by,omitempty"`
+	Workflow         string `url:"workflow,omitempty"`
+	PullRequestID    int    `url:"pull_request_id,omitempty"`
+	TriggerEventType string `url:"trigger_event_type,omitempty"`
+}
+
+func (f filter) String() string {
+	return fmt.Sprintf("- branch: %s\n- workflow: %s\n- pull request ID: %d\n- event type: %s", f.Branch, f.Workflow, f.PullRequestID, f.TriggerEventType)
+}
+
+type build struct {
+	Tag                     string    `json:"tag"`
+	Slug                    string    `json:"slug"`
+	Branch                  string    `json:"branch"`
+	Status                  int       `json:"status"`
+	CommitHash              string    `json:"commit_hash"`
+	StatusText              string    `json:"status_text"`
+	TriggeredAt             time.Time `json:"triggered_at"`
+	BuildNumber             int64     `json:"build_number"`
+	PullRequestID           *int64    `json:"pull_request_id"`
+	TriggeredWorkflow       string    `json:"triggered_workflow"`
+	PullRequestTargetBranch string    `json:"pull_request_target_branch"`
+}
+
+func (build build) generateFilter() filter {
+	f := filter{
+		Workflow: build.TriggeredWorkflow,
+		Branch:   build.Branch,
+		Before:   int(build.TriggeredAt.Unix()),
+	}
+	switch build.buildType() {
+	case buildTypeTag:
+		f.TriggerEventType = "tag"
+	case buildTypePullRequest:
+		f.TriggerEventType = "pull-request"
+		f.PullRequestID = int(*build.PullRequestID)
+	default: // website handles manual as push
+		f.TriggerEventType = "push"
+	}
+	return f
+}
+
+func (build build) buildType() string {
+	switch {
+	case build.Tag != "":
+		return buildTypeTag
+	case *build.PullRequestID > 0 || (build.PullRequestTargetBranch != ""):
+		return buildTypePullRequest
+	case build.CommitHash == "":
+		return buildTypeManual
```

Comment:
Based on comment in line 134 this does not happen.

Reply:
it is happening, the difference between the two logic is that the filter for the API doesn't allow me to set manual, but I need to filter locally for manual builds.